### PR TITLE
Add types

### LINKED ISSUE #4: Typescript typings for this plugin
Unlike many other plugins in the unified/remark/rehype ecosystem this one unfortunately doesn't provide Typescript typings. It would be nice if they could be provided, even though they will end up pretty short with only one exported function.

### PR DESCRIPTION
**Pull request by ChristianIvicevic**
Closes #4

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Minimum TypeScript Version: 3.0
+import {Plugin} from 'unified'
+
+/**
+ * Add `id`s to headings.
+ */
+declare const slug: Plugin
```
Comment (ChristianMurphy) [inline types/index.d.ts:7]: The plugin accepts no options, plugin with the default generic accepts one option

```suggestion
declare const slug: Plugin<[]>
```

would be more accurate.

**Review by ChristianMurphy — COMMENTED** [review]
Thanks @ChristianIvicevic!

File: types/index.d.ts (hunk 1 above)
Comment (ChristianIvicevic) [reply types/index.d.ts]: Ah, I didn't notice the index signature in the implicit default of the generic. I fixed that with the latest commit.

**Comment by ChristianIvicevic** [comment]
@ChristianMurphy Just a minor question on your opinion. The typings file does not necessarily need any `// Minimum TypeScript Version: 3.0` annotation right now. Do you want me to remove that as well?

**Comment by ChristianMurphy** [comment]
> The typings file does not necessarily need any `// Minimum TypeScript Version: 3.0` annotation right now. Do you want me to remove that as well?

Oh? DTSLint uses the comment to determine which versions of TypeScript to test on.
Removing the comment would probably cause the type tests to start failing.

File: types/tslint.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "dtslint/dtslint.json",
+  "rules": {
+    "no-redundant-jsdoc": false,
```
Comment (ChristianMurphy) [inline types/tslint.json:4]: there don't appear to be any jsdoc `@` annotations, this probably isn't needed.

```suggestion
```

File: package.json [hunk 3]
```diff
@@ -35,13 +36,15 @@
...
     "browserify": "^16.0.0",
+    "dtslint": "^4.0.5",
     "nyc": "^15.0.0",
     "prettier": "^2.0.0",
     "rehype": "^11.0.0",
     "remark-cli": "^8.0.0",
     "remark-preset-wooorm": "^7.0.0",
     "tape": "^5.0.0",
     "tinyify": "^2.0.0",
+    "unified": "^9.2.0",
```
Comment (ChristianMurphy) [inline package.json:47]:
```suggestion
    "unified": "^9.0.0",
```

File: package.json [hunk 4]
```diff
@@ -35,13 +36,15 @@
   },
   "devDependencies": {
     "browserify": "^16.0.0",
+    "dtslint": "^4.0.5",
```
Comment (ChristianMurphy) [inline package.json:39]: for consistency it's good to keep the semver range as wide as possible
```suggestion
    "dtslint": "^4.0.0",
```

**Review by ChristianMurphy — APPROVED** [review]

**Review by remcohaszing — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
released, thanks @ChristianIvicevic!

**Pull request merged**